__tests__/app.test.js

```diff
@@ -177,4 +175,55 @@ describe("GET /api/articles/:article_id/comments",()=>{
         expect(body.msg).toBe('bad request')
     })
    })
+})
+describe("POST /api/articles/:article_id/comments",()=>{ 
+    
+    test("responds with 201 and with the new comment", ()=>{
+        const newComment =[{
+            username: 'lurker',
+            body: 'adding new comment'
+            } ]
+       
+        return request(app)
+            .post("/api/articles/2/comments")
+            .send(newComment)
+            .expect(201)
+            .then(({body})=>{
+                expect(body).toEqual({
+                    comment_id:19,
+                    body: 'adding new comment',
+                    votes: 0,
+                    author: "lurker",
+                    article_id: 2,
+                    created_at: expect.any(String)
+                })
+            })
+        
+    })
+    test("responds with 404 not found when there is no article found with given id",()=>{
+        const newComment =[{
+            username: 'lurker',
+            body: 'adding new comment'
+            } ]
+        return request(app)
+    .post('/api/articles/55/comments')
+    .send(newComment)
+    .expect(404)
+    .then(({body})=>{
+        expect(body.msg).toBe('article not found')
+    })
+    })
+    test("responds with 400 when given id is NaN",()=>{
+        const newComment =[{
+            username: 'lurker',
+            body: 'adding new comment'
+            } ]
+        return request(app)
+    .post('/api/articles/two/comments')
+    .send(newComment)
+    .expect(400)
+    .then(({body})=>{
+        expect(body.msg).toBe('bad request')
+    })
+    })
```

Comment:
Great testing so far - there's a couple more tests to consider here:

- [ ] Status 201: Creates new comment when there are extra keys in request body, ignoring extra keys
- [ ] Status 400: Bad request body, i.e. missing username/body properties
- [ ] Status 404: Username does not exist in database